From 579e707b3a16aa5aa4fe4770429e7f72229f449e Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 5 Dec 2016 17:38:23 +0800
Subject: [PATCH 1369/5242] MLK-13607-1 ARM: dts: imx7ulp-evk: add gpio
 backlight support.

commit  93fa87b70ad6cd82fac4b34cdce9fec01fff128e from
https://source.codeaurora.org/external/imx/linux-imx.git

The mipi dsi panel on imx7ulp-evk board will use gpio to control
lcd backlight.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 763dbfe8a24dd11827f5024327b400f73917d118)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk.dts |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index 5505046..8b27d61 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -31,6 +31,15 @@
 		reg = <0x60000000 0x40000000>;
 	};
 
+	backlight {
+		compatible = "gpio-backlight";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_backlight>;
+		gpios = <&gpio3 2 GPIO_ACTIVE_HIGH>;
+		default-on;
+		status = "okay";
+	};
+
 	regulators {
 		compatible = "simple-bus";
 		#address-cells = <1>;
@@ -159,6 +168,12 @@
 			>;
 		};
 
+		pinctrl_backlight: backlight_grp {
+			fsl,pins = <
+				IMX7ULP_PAD_PTF2__PTF2		0x20100
+			>;
+		};
+
 		pinctrl_lpi2c5: lpi2c5grp {
 			fsl,pins = <
 				IMX7ULP_PAD_PTC4__LPI2C5_SCL       0x527
-- 
1.7.9.5

